analyzers/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs

```diff
@@ -72,7 +72,7 @@ private static void ReportDiagnostic(ReportingContext reportingContext)
             }
 
             // ... but for compatibility purposes we need to keep handling the old-fashioned way. Old SonarLint can be used with latest NuGet.
-            if (SonarAnalysisContext.AreAnalysisScopeMatching(reportingContext.Compilation, new[] { reportingContext.Diagnostic.Descriptor }) &&
+            if (reportingContext.Compilation.AreAnalysisScopeMatching(new[] { reportingContext.Diagnostic.Descriptor }) &&
```

Comment:
It is either AreAnalysisScopesMatching or IsAnalysisScopeMatching. looking at the body of the method I would prefer the second.